src/App.tsx

```diff
@@ -1,10 +1,64 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useMemo, useEffect, useState } from 'react';
+import cn from 'classnames';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos, USER_ID } from './api/todos';
+import { ErrorNotification } from './components/Errors';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { ErrorText } from './types/ErrorText';
+import { Filter } from './components/Filters';
+import { Filters } from './types/filters';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState<Filters>(Filters.all);
+
+  const hideError = () => {
+    setErrorMessage('');
+  };
+
+  useEffect(() => {
+    if (!errorMessage) {
+      return;
+    }
+
+    const t = setTimeout(() => setErrorMessage(''), 3000);
+
+    return () => clearTimeout(t);
+  }, [errorMessage]);
+
+  useEffect(() => {
+    hideError();
+    setIsLoading(true);
+    getTodos()
+      .then(todosFromServer => setTodos(todosFromServer))
+      .catch(() => setErrorMessage(ErrorText.load))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const itemsLeft = useMemo(
+    () => todos.filter(t => !t.completed).length,
+    [todos],
+  );
+
+  const hasTodos = todos.length > 0;
+  const allCompleted = hasTodos && itemsLeft === 0;
+
+  const filteredTodos = useMemo(() => {
+    switch (filter) {
+      case Filters.active:
```

Comment:
enums are usually capitalised 
```suggestion
      case Filters.Active:
```